g2p_entitlement_cash/models/entitlement_manager.py

```diff
@@ -291,8 +291,26 @@ def _validate_entitlements(self, cycle, offset=0, limit=None):
             limit=limit,
         )
         err, message = self.approve_entitlements(entitlements)
+        if err:
+            cycle.message_post(body=_(message))
+            cycle.write({"validate_async_err": True, "state": "to_approve"})
+        else:
+            cycle.write({"validate_async_err": False})
         return err, message
 
+    def mark_job_as_done(self, cycle, msg):
+        if cycle.validate_async_err:
+            super().mark_job_as_done(
+                cycle,
+                _(
+                    "Entitlements are not approved due to some issues. Kindly address the issue."
+                ),
+            )
+        else:
+            super().mark_job_as_done(cycle, msg)
+
+        return
```

Comment:
```suggestion
        if cycle.validate_async_err:
            msg = _("Entitlements are not approved due to some issues. Kindly address the issue.")
        return super(). mark_job_as_done(cycle, msg)
```
I think this will made the code more readable

Reply:
I agree with @nhatnm0612